timeeval/remote_configuration.py

```diff
@@ -13,24 +13,84 @@
 
 @dataclass
 class RemoteConfiguration:
+    """This class holds the configuration for distributed TimeEval.
+
+    TimeEval uses a :obj:`dask.distributed.SSHCluster` to distribute the evaluation tasks to multiple compute nodes.
+    Please read the Dask documentation carefully and then use the constructor arguments to setup a TimeEval cluster.
+
+    Parameters
+    ----------
+    scheduler_host : str
+        IP address or hostname for the :obj:`distributed.Scheduler`.
+        This node will be responsible to coordinate the cluster.
+        The scheduler does not perform any evaluations.
+    scheduler_port : int
+        Port for the scheduler.
+    worker_hosts  : List[str]
+        List of IP address or hostnames for the :obj:`distributed.Worker`.
+        These nodes will execute the evaluation tasks.
+    remote_python : str
+        Path to the Python-executable.
+        If you set up all your nodes in the same way, the default is fine.
+    kwargs_overwrites : dict
+        Use this option to overwrite any configuration options of the :obj:`~dask.distributed.SSHCluster`.
+
+        .. warning::
+            Only use if you know what you are doing!
+
+    dask_logging_file_level : str
+        Logging level for the file-based Dask logger.
+    dask_logging_console_level : str
+        Logging level for the console-based Dask logger.
+    dask_logging_filename : str
+        Name of the Dask logging file without any parent paths.
+        Each node will write its own logging file and TimeEval will automatically postfix the filenames with the hostname and
+        place the Dask logging files into the ``results_path``.
+
+    Examples
+    --------
+    Two-node cluster where the first node hosts the scheduler but also takes part in the evaluation:
+
+    >>> from timeeval import TimeEval, RemoteConfiguration
+    >>> config = RemoteConfiguration(scheduler_host="192.168.1.1", worker_hosts=["192.168.1.1", "192.168.1.2"])
+    >>> TimeEval(dm=..., datasets=[], algorithms=[], distributed=True, remote_config=config)
+    """
     scheduler_host: str = DEFAULT_SCHEDULER_HOST
     scheduler_port: int = DEFAULT_DASK_PORT
     worker_hosts: List[str] = field(default_factory=lambda: [])
-    remote_python: str = sys.executable
+    remote_python: str = field(default_factory=lambda: sys.executable)
```

Comment:
This should also work: `field(default=sys.executable)`

Reply:
Yes, it does and `remote_python: str = sys.executable` also works, but when building the documentation the local string is then used as default value. E.g., when building the documentation on my machine:

![image](https://user-images.githubusercontent.com/10573700/182565599-aede214f-f0f5-484c-9612-4e26deef951a.png)

This is misleading because we actually use the python executable of the executing machine and not `/home/sebastian/.../python`.

---

The underlying question is, if we should change source code to make the documentation clearer?

Reply:
ok